src/common/catalog/catalog-entity.ts

```diff
@@ -59,6 +66,18 @@ export abstract class CatalogCategory extends EventEmitter {
   public getId(): string {
     return `${this.spec.group}/${this.spec.names.kind}`;
   }
+
+  public async onLoad(): Promise<void> {
+    for( const listener of this.listeners("onLoad")) {
```

Comment:
[nit] if linter is happy with this extra space, I am fine too.
```suggestion
    for(const listener of this.listeners("onLoad")) {
```

Reply:
should it be 
```
Should it be:

```suggestion
    for (const listener of this.listeners("onLoad")) {
```

Reply:
These are now removed.